### PR TITLE
monitoring: add test stats to database

### PR DESCRIPTION
**Pull request by sodar**
~~Blocked by #29~~

### CODE REVIEW

**Comment by sodar** [comment]
recheck no bug

**Comment by sodar** [comment]
recheck no bug

File: monitoring/tests/test_build_stats_collector.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        self.example_test_stats_collector = MagicMock()
+        self.example_test_stats_collector.collect = MagicMock(return_value=self.example_test_stats)
+
+    def test_collecting_works(self):
+        collector = BuildStatsCollector(build_stats_collector=self.example_build_stats_collector,
+                                        test_stats_collector=self.example_test_stats_collector)
+        build_stats = collector.collect()
+        self.assertIsNotNone(build_stats)
+
+        self.assertEqual(build_stats.build_id, self.example_build_stats.build_id)
```
Comment (jacek-izykowski) [inline monitoring/tests/test_build_stats_collector.py]: Uncle Bob wouldn't be impressed... But we don't complain.
MK suggests helper assertion encapsulating these assertions.

File: monitoring/tests/test_collectors.py [hunk 2]
```diff
@@ -102,5 +106,144 @@ def test_stages_stats(self):
...
+            m.get('http://1.2.3.4/build/1/test2.xml', status_code=404)
+
+            collector = XmlReportCollector(url='http://1.2.3.4/build/1/test.json')
+            with self.assertRaises(InvalidResponseCodeError):
+                collector.collect()
+
+    def test_raises_error_when_some_fields_do_not_exist(self):
+        with requests_mock.mock() as m:
+            response_text = '{ "xml_reports": ["report.xml"], "html_report": "report.html" }'
+            m.get('http://1.2.3.4/build/1/test.json', text=response_text)
```
Comment (jacek-izykowski) [inline monitoring/tests/test_collectors.py]: Multiple constant URLs (e.g. 'http://1.2.3.4/build/1/test.json') are repeated in the code.

File: monitoring/collectors/xml_report_collector.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import re
```
Comment (jacek-izykowski) [inline monitoring/collectors/xml_report_collector.py]: Do we need it?

**Comment by jacek-izykowski** [comment]
Testing GitHub...

**Pull request merged**